torch_xla/csrc/reduction.cpp

```diff
@@ -422,6 +422,24 @@ xla::XlaOp BuildAll(xla::XlaOp input, absl::Span<const xla::int64> dimensions,
   xla::XlaOp result =
       xla::Reduce(input, init_value, CreateAllComputation(shape.element_type()),
                   dimensions);
+  bool all_reduce_dims_size_one = true;
```

Comment:
I think it is a bit more elegant to use `std::any_of()` or `std::all_of()` for something like this.